code/datums/datumvars.dm

```diff
@@ -592,6 +593,35 @@ body
 			if(usr.client)
 				usr.client.cmd_assume_direct_control(M)
 
+		else if(href_list["offer_control"])
+			if(!check_rights(0))	return
+
+			var/mob/M = locate(href_list["offer_control"])
+			if(!istype(M))
+				usr << "This can only be used on instances of type /mob"
+				return
+			M << "Control of your mob has been offered to dead players."
+			log_admin("[key_name(usr)] has offered control of [M.real_name] to ghosts.")
+			message_admins("<span class='notice'>[key_name(usr)] has offered control of [M.real_name] to ghosts</span>")
+			var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as [M.real_name]?", "pAI", null, FALSE, 100)
+			var/mob/dead/observer/theghost = null
+
+			if(candidates.len)
+				for(var/mob/j in candidates)
+					if(!j || !j.client)
+						candidates.Remove(j)
+						continue
+					theghost = j
+					M << "Your mob has been taken over by a ghost!"
+					usr << "[theghost.key] has taken over [M.real_name]."
+					message_admins("<span class='notice'>[key_name(theghost)] has taken control of [M.real_name]</span>")
+					M.ghostize()
+					M.key = theghost.key
+					break

```

Comment:
Because you use a for loop + break after 1 iteration to find a ghost, you will always pick ```candidates[1]```
this means the first person to push the button will get the role, which is ok, but unfair, just use a ```pick()```.